### ISSUE TITLE
return more contact properties from UserLogin webservice

### ISSUE THREAD

**Issue by davisagli** [washingtontrails/vms#26]
@mscholtz Charlie wants the registration form to be prefilled with contact info from Salesforce for logged in users. How hard is it to make the UserLogin webservice return these additional properties of the Contact?
- phone
- phoneType (Home Phone, Mobile/Cell, Work Phone)
- street/city/state/zip

**Comment by mscholtz** [comment]
Not hard at all.  Don't we have a separate story for this?  I know it's
been discussed previously.

On Wed, Feb 18, 2015 at 8:51 PM, David Glick <notifications@github.com>
wrote:

> @mscholtz <https://github.com/mscholtz> Charlie wants the registration
> form to be prefilled with contact info from Salesforce for logged in users.
> How hard is it to make the UserLogin webservice return these additional
> properties of the Contact?
>
>    - phone
>    - phoneType (Home Phone, Mobile/Cell, Work Phone)
>    - street/city/state/zip
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/26>.
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by davisagli** [comment]
I don't see a separate story for it, but it's arguably part of the "logged in users register" story from iteration 2. Charlie asked Sally and Karen earlier today about prefilling these and whether it should be a new story or a ticket, and I'm thinking it's probably small enough to be a ticket (though I don't have a sense of how much time Percolator has set aside for working on tickets).

**Comment by mscholtz** [comment]
It's fine as a ticket, then.

On Wed, Feb 18, 2015 at 9:44 PM, David Glick <notifications@github.com>
wrote:

> I don't see a separate story for it, but it's arguably part of the "logged
> in users register" story from iteration 2. Charlie asked Sally and Karen
> earlier today about prefilling these and whether it should be a new story
> or a ticket, and I'm thinking it's probably small enough to be a ticket
> (though I don't have a sense of how much time Percolator has set aside for
> working on tickets).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/26#issuecomment-75002265>.
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by mscholtz** [comment]
@ninetaLogik do you feel comfortable working on this one?  It's modifying a webservice I wrote, but it's just adding a few fields to what the ws returns.  Take a look at the code, and let me know what you think? (Please make sure you refresh before starting work.)

**Comment by ninetaLogik** [comment]
@mscholtz I think the queries I need to modify are in find_or_create_contact correct? The matchingCon and matchingCons queries I believe.

**Comment by ninetaLogik** [comment]
@davisagli We can return these fields for the contact :Phone, HomePhone, MobilePhone, MailingStreet, MailingCity, MailingPostalCode, MailingState 
Can you use these to get the Phone and Phone Type?

**Comment by WaBirder57** [comment]
Can't you use the field npe01__PreferredPhone__c to get the "type" of the phone number contained in the field "Phone"?

**Comment by ninetaLogik** [comment]
We can, but I wasn't sure if you were using that field. If you are, that's great I'll go ahead and return that as well. I noticed that when we create a new Contact from the user login ws we don't set the Preferred Phone.

**Comment by davisagli** [comment]
The preferred phone is not set by the user login webservice because there's no phone stored in Plone, so we don't have one to pass to the webservice.

We should however set npe01_PreferredPhone__c when the registration webservice is called, if we don't do that already.

Please add all 3 phone numbers and npe01_PreferredPhone__c to the login webservice response. I'll use the field indicated by npe01_PreferredPhone__c if it's not empty, otherwise pick whatever field has a value.

**Comment by ninetaLogik** [comment]
@davisagli : WRT : We should however set npe01_PreferredPhone__c when the registration webservice is called, if we don't do that already.

So whatever Type they pick for the Phone at registration we set the preferred phone to that, correct?

**Comment by davisagli** [comment]
I think only set it to the type picked during registration if there isn't a preferred phone set yet.

**Comment by ninetaLogik** [comment]
@davisagli I have made the changes to the web service to return the address and phone information. Also, modified the registration web service to set the preferred phone flag if not already set. Could you please let me know if these work for you.

**Comment by davisagli** [comment]
Looks good, @ninetaLogik.

@WaBirder57 the pyramid app should now fetch address, phone, and phone type from Salesforce when you log in, and prefill the registration form.

**Comment by WaBirder57** [comment]
Outstanding, works like a charm.  Thank you both.
